src/Microsoft.Health.Fhir.Core/Exceptions/MemberMatchMatchingException.cs

```diff
@@ -0,0 +1,27 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Exceptions
+{
+    /// <summary>
+    /// Exception to indicate no match or multiple matches for member-match operation.
+    /// </summary>
+    public class MemberMatchMatchingException : FhirException
+    {
+        public MemberMatchMatchingException(string message)
+            : base(message)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(message), "Exception message should not be empty");
+
+            Issues.Add(new OperationOutcomeIssue(
+                    OperationOutcomeConstants.IssueSeverity.Error,
+                    OperationOutcomeConstants.IssueType.Forbidden,
```

Comment:
Should this be `IssueType.Invalid` instead of `Forbidden` (since this is not a permissions related issue?)